integration/setup/tenant_cluster.go

```diff
@@ -295,3 +338,503 @@ func ensureCertConfigsInstalled(ctx context.Context, id string, config Config) e
+				        ],
+				        "name": "tuommaki",
+				        "sshAuthorizedKeys": [
+				          "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQCu2V1p0cZ2N4ucug9LQMB+YMg9AQ5+aZQTdDTZ7oBuEcBuGtdnSSbcxj1lHoMYvhz6ugFVolkusRnZSakZY/XPVlwIHC56TWWrJ0hJ4sQEzCqVSHx0ZBHaMZepxCz7KSh/4KjtZFyaBC9SFwUo7kGgBYoFdClhxZsmfMsk0RneY8FjWme/cwXSaEGdaaTyOA52UOCg6Ax3nnE/gAJBsL8HgI17bFjj8og6TdPoP+33wujGHFORy8HF/m6p1I2Nm9Mp+gkG6PzdkWbF7UFci5uYHXy5IEu6uGzEPQiB5BjgfVIvZyH3VfKxmG1T2yyp4/qDQOmkjlIahpPyI00Y3SWAab7MdQXJ2hTgWFo/NP+AEdd45+PrSvTMy2k5bVl9GMntP+z+9oAhwH8OStSCJ0GBGlVG89fd0vFV1XVmLPwS8XhuhAoU1KRt6/Hc8cs7uSUiKOTY8Xn6VNUozxK137QpHBb81jU7OCcmopF9dlqoV6m18iZK1NjP4+FFxUyi5O4HI6aFrZXf7Cw5G9C8EXML3qLIMxd2pIJsu8QTw/5kC7sBtmFY/5RqW0TZ5hWuyGSuFcRan5E08Qct5rGAQ6QjJ9rZqQUPeJFcN6gEvGUam0XdeziZD6lPFUDkte9y653lIrPqBoSbsJuk/FJU/+RTSYEl+VCmaac3ru6jYV6M8w== tuomas"
+				        ],
+				        "shell": "/bin/bash"
+				      },
+				      {
+				        "groups": [
+				          "sudo",
+				          "docker"
+				        ],
+				        "name": "theo",
+				        "sshAuthorizedKeys": [
+				          "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDo5PU8w8RDWYDd8SSIKOiYJCeN398PAEJApFGpPWprewBmiDGGMHHDIDV//QR+o5MhR2hBmJ+Pw1K/Edr4u0cfGlIIb6lSVdha+jDEp0l1PtqyQzubzTH3y/RDzxCakAa419N1G3DwzJHkWBxbpyqx7i/DOYcxgQP4vCGvYgkvuOkQYCNk2hfuXAl/Aucv3JXlsuNktyXQ6XKf+Twa2Bg8jIAaYUNGqgKgzMcsCElE55bxVuYeXl441CzD2fdHmXyGwo6nefN7PZ790SxQzkGM8wBpESgc7U4IPUY/dnsn4yQBYw2meontHWGLmZjrvEYxoS7Uv4o8BX8cScgVZUhRojHvNWBBcwOx+hhuPbqoqdc8IFXQLHTTa/szvY9gwlipBejj6nJrRpl3Kxw+EX4QP/loDxkykWQUoByU479Z6/gOtgAkPOe8xZblny6r3uCZyUlaYR9ht2aOEbH8bLuYBaDTPvunMIH/RSgbNxys/Dss3ZC7MJgXtoaSpb/AGdqv1Uj4AdNJJm8544AFhmR/Tky0rms3NpaSEiwO+E5ZIJiBevqPbGWkodbfKM6uydS+wrqHOR+zTNLuwhTHVnNRZ//ePBMptnzR1qbuMmaEgmqMM1HjFflUUVdDLFK6TxcdU3YPJnnWlwGk2bCjlamGjHx0hvoJVSY+Oap6o23cJQ== theo@giantswarm.io"
+				        ],
+				        "shell": "/bin/bash"
+				      }
+				    ]
+				  },
+				  "storage": {},
+				  "systemd": {}
+				}
+			`))),
+		}
+
+		_, err := config.AWSClient.EC2.RunInstances(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "created bastion instance")
+	}
+
+	return nil
+}
+
+func ensureBastionHostDeleted(ctx context.Context, clusterID string, config Config) error {
+	var err error
+
+	var instanceID string
+	{
+		config.Logger.LogCtx(ctx, "level", "debug", "message", "finding bastion instance id")
+
+		i := &ec2.DescribeInstancesInput{
+			Filters: []*ec2.Filter{
+				{
+					Name: aws.String("tag:giantswarm.io/cluster"),
+					Values: []*string{
+						aws.String(clusterID),
+					},
+				},
+				{
+					Name: aws.String("tag:giantswarm.io/instance"),
+					Values: []*string{
+						aws.String("e2e-bastion"),
+					},
+				},
+			},
+		}
+
+		o, err := config.AWSClient.EC2.DescribeInstances(i)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if len(o.Reservations) == 0 {
+			return microerror.Maskf(notExistsError, "master instance")
```

Comment:
Usually we have `notFoundError` but I don't see it being matched anyway. I would skip this `if`.

Reply:
As mentioned in the comment about the error matcher, this is copied from the tccp resource. The code is identical. I would rather like to have the copied and identical code than snowflakes, because we can more easily abstract it later on.

Reply:
Okay. Confusing but fine with me.